comet/broker.py

```diff
@@ -31,6 +33,67 @@
 app.config.RESPONSE_TIMEOUT = 120
 
 
+request_id = contextvars.ContextVar("request_id", default=0)
+
+
+lock_datasets = None
+lock_states = None
+
+waiting_datasets = {}
+waiting_states = {}
+
+
+class RequestFormatter(logging.Formatter):
+    """Logging formatter that adds a request_id.
+
+    Parameters
+    ----------
+    request_id : ContextVar
+        A context variable that contains the request ID.
+    format : str
+        A format string which represents how the log should be formatted.
+    """
+
+    def __init__(self, request_id=0, format=format):
+        self.request_id = request_id
+        super().__init__(format)
+
+    def format(self, record):
+        """Return a formatted string for the log.
+
+        Set the record.request_id.
+
+        Parameters
+        ----------
+        record : dict
+            A set of relevant attributes of the logged message
+            that are parsed into the format string.
+        """
+        record.request_id = self.request_id.get()
+        return logging.Formatter.format(self, record)
+
+
+logger = logging.getLogger(__name__)
+syslog = logging.StreamHandler()
+# reqfilter = RequestFilter(request_id)
+# formatter = RequestFormatter(request_id)
+# formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+formatter = RequestFormatter(
+    request_id,
+    "%(asctime)s [%(process)d] [%(name)s:%(levelname)s] [request=%(request_id)s] %(message)s",
+)
+syslog.setFormatter(formatter)
+logging.getLogger("comet").setLevel(logging.DEBUG)
+# logging.getLogger("").addFilter(reqfilter)
+logging.getLogger("").addHandler(syslog)
```

Comment:
When I merged @jrs65's branches into this, I saw you had passed `"comet"` to `getLogger` and Richard removed it. I don't know what we want in the end. Do you?

Reply:
I do!

Richard wants this to get associated with the root logger. Which means that the correct thing to do is to pass `''` to `getLogger`.

Reply:
(I agree with associating this with the root logger. =))